### ISSUE TITLE
RandLA torch results

### ISSUE THREAD

**Issue by saltoricristiano** [intel-isl/Open3D-ML#168]
Hi, 
I run the pretrained RandLA torch model for SemanticKITTI on the validation split of SemanticKITTI and I got the following numbers:
```
test acc: 0.6835311938123804
test iou: 0.4933693851882593
```
While the reported mIoU is of 52.8. Are these numbers referred to the test split? If yes, which are the results on the validation split? For validating on the valid split I simply changed the evaluated split in your evaluation tutorial code.
Thanks in advance for any help!

**Comment by YilingQiao** [comment]
Hi! Yes, the reported result is for the test set. You can also run on the test set and submit it to SemanticKITTI leaderboard and see the results.

How did you get the `0.493`? Did you change the test split in the config file? Can you please share your command with me so that I can also run it? Thanks!

**Comment by saltoricristiano** [comment]
Hi @YilingQiao, thanks for answering!
Ok so the numbers I'm getting are on the validation set and not on the test set as yours. 
To get these numbers I only changed these lines:

- I changed the dataset split loaded on TorchDataloader:
```
test_split = TorchDataloader(dataset=dataset.get_split('validation'),
                                     preprocess=model.preprocess,
                                     transform=model.transform,
                                     use_cache=dataset.cfg.use_cache,
                                     shuffle=False)
```

- the same for datset_split :
`datset_split = self.dataset.get_split('validation')`

After that, I run the inference on the validation set following your tutorial so simply `pipeline.run_test()`

I noticed other three things that may be considered:

**1.** the checksum of RandLA torch (_randlanet_semantickitti_202009090354utc.pth_) I get are different from the ones reported in the .txt. I tried to download multiple times and I got always the same code which is _9b353b6bccaa2cbff9f71ad0ab008baa018f050d0cc30ed75a3d4bb8d264c6e6_ while the reported one is _44e916396f491cf6fdabf8dcbe03736f_ . Can this be the problem? (I would expect to have a non working model in case of a corrupted .pth, isn't it?)

**2.** while inferencing either on validation or test split, I was getting  this warning:
`UserWarning: Mixed memory format inputs detected while calling the operator. The operator will output channels_last tensor even if some of the inputs are not in channels_last format. (Triggered internally at  /pytorch/aten/src/ATen/native/TensorIterator.cpp:924.)` and I solved it by putting a `.contiguous()` on both `f_pc` and `shortcut` here https://github.com/intel-isl/Open3D-ML/blob/5549444280ca02f3881cdf06f15c1ed06efb7389/ml3d/torch/models/randlanet.py#L433

**3.** while inferencing, I get these warning due to the presence of `Nan` in the predictions (avoided by using np.nanmean() by default) :
 ```
 /opt/conda/lib/python3.6/site-packages/open3d/_ml3d/torch/modules/metrics/semseg_metric.py:130: RuntimeWarning: 
 invalid value encountered in long_scalars
  iou = iou / (pred_mask | labels_mask).sum()
 ```
and 
```
opt/conda/lib/python3.6/site-packages/open3d/_ml3d/torch/modules/metrics/semseg_metric.py:151: RuntimeWarning: invalid value encountered in long_scalars
  per_class_accuracy /= label_mask.sum()
```

**Comment by YilingQiao** [comment]
Thanks very much for your detailed description and other suggestions! I just started running with your configuration and will get back to you in one or two days.

**Comment by YilingQiao** [comment]
Hi, I ran on the `dev` branch, which will be the next release in the next week, and got the iou of the validation set is `52.0`. 

I think the reason you get a lower IoU in the previous version is that the IoU was the mean of the IoUs of each point cloud. But in real testing, we will accumulate all the results of different point clouds into a big confusion matrix and count the IoU based on the single matrix.

Due to the unbalanced label distribution, the IoU will be affected significantly by the disturbance of classes with fewer points, and the IoU will be higher and more stable if there are enough samples, which is the case where you accumulate them into a big confusion matrix. And the new release will compute the mIoU in this way.

Thanks for your suggestion in point 1, 2, 3! We will update them in a new PR.

**Comment by saltoricristiano** [comment]
Hi @YilingQiao , I tried with the confusiont mtx in the `dev` branch as you said and I've been able to reproduce the numbers you said! Thank you very much for the help!

**Comment by saltoricristiano** [comment]
Hi @YilingQiao, following your suggestion and using the confusion mtx of the `dev` branch I've been able to get the same numbers! Thank you for the help!

**Comment by Hao-HUST** [comment]
> Hi,
> I run the pretrained RandLA torch model for SemanticKITTI on the validation split of SemanticKITTI and I got the following numbers:
> 
> ```
> test acc: 0.6835311938123804
> test iou: 0.4933693851882593
> ```
> 
How did you evaluate the pre-trained RandLA torch model on the validation split of SemanticKITTI? Could you please release your modified code about that?
